tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -580,19 +584,30 @@ private static void EnableCallSiteInstrumentations(InstrumentationCategory categ
             // enabled and IAST is disabled. We don't expect RASP only instrumentation to be used in the near future.
 
             var isIast = categories.HasFlag(InstrumentationCategory.Iast);
+            var debugMsg = (isIast && raspEnabled) ? "IAST/RASP" : (isIast ? "IAST" : "RASP");
 
             if (isIast || raspEnabled)
             {
-                string[] inputAspects = null;
-
-                inputAspects = isIast ? AspectDefinitions.GetAspects() : AspectDefinitions.GetRaspAspects();
+                Log.Debug("Registering {DebugMsg} Callsite Dataflow Aspects into native library.", debugMsg);
+                int aspects = 0;
+                if (_useEmbeddedDefinitions)
+                {
+                    string platform = ConfigTelemetryData.ManagedTracerTfmValue;
+                    if (!isIast) { platform += "_Rasp"; }
 
-                if (inputAspects != null)
+                    aspects = NativeMethods.InitEmbeddedCallSiteDefinitions(platform, (uint)categories);
+                }
+                else
                 {
-                    var debugMsg = (isIast && raspEnabled) ? "IAST/RASP" : (isIast ? "IAST" : "RASP");
-                    Log.Debug("Registering {DebugMsg} Callsite Dataflow Aspects into native library.", debugMsg);
+                    string[] inputAspects = isIast ? AspectDefinitions.GetAspects() : AspectDefinitions.GetRaspAspects();
+                    if (inputAspects != null)
+                    {
+                        aspects = NativeMethods.RegisterIastAspects(inputAspects);
+                    }
+                }
 
-                    var aspects = NativeMethods.RegisterIastAspects(inputAspects);
+                if (aspects > 0)
```

Comment:
If aspects returns 0 here, it means something went wrong I think? Maybe we should log that as an error here, so that it's recorded in the managed logs (and telemetry)?

Reply:
Fixed in 39a7405fa42690cd86b202dd7292f4083b63456e